src/scripts/main.js

```diff
@@ -1,3 +1,256 @@
 'use strict';
 
-// write your code here
+const gameContainer = document.querySelector('.container');
+const startButton = gameContainer.querySelector('.button');
+const cells = gameContainer.querySelectorAll('.field-cell');
+const messageStart = gameContainer.querySelector('.message-start');
+const messageWin = gameContainer.querySelector('.message-win');
+const messageLose = gameContainer.querySelector('.message-lose');
+const gameScore = gameContainer.querySelector('.game-score');
+const gameRow = gameContainer.querySelectorAll('tr');
+let movePossible = false;
+
+function generateRandom() {
+  const randomI = Math.floor(Math.random() * cells.length);
+
+  if (cells[randomI].textContent === '') {
+    const randomNum = Math.random() < 0.1 ? '4' : '2';
+
+    cells[randomI].textContent = randomNum;
+    cells[randomI].classList.add('field-cell--' + randomNum);
+  } else {
+    generateRandom();
+  }
+}
+
+startButton.addEventListener('click', () => {
+  if (startButton.classList.contains('start')) {
+    generateRandom();
+    generateRandom();
+    messageStart.classList.add('hidden');
+    startButton.classList.remove('start');
+    startButton.classList.add('restart');
+    startButton.textContent = 'Restart';
+  } else if (startButton.classList.contains('restart')) {
+    cells.forEach((cell) => {
+      cell.textContent = '';
+      cell.className = 'field-cell';
+    });
+    messageStart.classList.remove('hidden');
+    startButton.classList.remove('restart');
+    startButton.classList.add('start');
+    startButton.textContent = 'Start';
+    gameScore.textContent = '0';
+    messageLose.classList.add('hidden');
+    messageWin.classList.add('hidden');
+  }
+});
+
+function updateCells() {
+  cells.forEach((cell) => {
+    if (cell.textContent === '') {
+      cell.className = 'field-cell';
+    } else {
+      cell.className = 'field-cell';
+      cell.classList.add('field-cell--' + cell.textContent);
+    }
+  });
+}
+
+document.addEventListener('keydown', (e) => {
+  if (e.key === 'ArrowLeft') {
+    slideLeft();
+    checkGameOver();
+  }
+
+  if (e.key === 'ArrowRight') {
+    slideRight();
+    checkGameOver();
+  }
+
+  if (e.key === 'ArrowUp') {
+    slideUp();
+    checkGameOver();
+  }
+
+  if (e.key === 'ArrowDown') {
+    slideDown();
+    checkGameOver();
+  }
```

Comment:
- move 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown' to the constants 
- please use switch case hare 
- function checkGameOver(), try to call this function only one time, but with the condition